Indicators/IndicatorExtensions.cs

```diff
@@ -62,6 +64,44 @@ public static bool Update(this IndicatorBase<IndicatorDataPoint> indicator, Date
         }
 
         /// <summary>
+        /// Creates a new CompositeIndicator such that the result will be average of a first indicator weighted by a second one
+        /// </summary>
+        /// <param name="value">Indicator that will be averaged</param>
+        /// <param name="weight">Indicator that provides the average weights</param>
+        /// <param name="period">Average period</param>
+        /// <returns>Indicator that results of the average of first by weights given by second</returns>
+        public static CompositeIndicator<IndicatorDataPoint> WeightedBy<T, TWeight>(this IndicatorBase<T> value, TWeight weight, int period)
+            where T : BaseData
+            where TWeight : IndicatorBase<IndicatorDataPoint>
+        {
+            var x = new WindowIdentity(period);
+            var y = new WindowIdentity(period);
+            var numerator = new Sum("Sum_xy", period);
+            var denominator = new Sum("Sum_y", period);
+
+            value.Updated += (sender, consolidated) =>
+            {
+                x.Update(consolidated);
+                if (x.Samples == y.Samples)
```

Comment:
We could, alternatively, only update the numerator when `x.Current.EndTime` equals `y.Current.EndTime`. Would that be better? It won't solve the resolution, but it would accept data with the same resolution but that are ready with different samples.